src/components/SuggestionTags/hooks/useSuggestionTags.tsx

```diff
@@ -0,0 +1,208 @@
+import useQueryTokenMetadata from './useQueryTokenMetadata';
+
+const HIGH_PRICE_IMPACT = 5; // 5%
+const MINIMUM_THRESHOLD_FOR_DCA = 1_000; // 1,000 USD, not USDC
+
+const FREEZE_AUTHORITY_IGNORE_LIST = [USDC_MINT.toString(), USDT_MINT.toString(), JLP_MINT.toString()];
+
+export const useSuggestionTags = ({
+  fromTokenInfo,
+  toTokenInfo,
+  quoteResponse,
+}: {
+  fromTokenInfo: TokenInfo | null | undefined;
+  toTokenInfo: TokenInfo | null | undefined;
+  quoteResponse: QuoteResponse | undefined;
+}) => {
+  const { data: tokenMetadata } = useQueryTokenMetadata({ fromTokenInfo, toTokenInfo });
+  const { data: lstApy } = useLstApyFetcher(['JupSOL']);
+  const birdeyeInfo = useBirdeyeRouteInfo();
+  const { tokenPriceMap } = useUSDValue();
+
+  const listOfSuggestions = useMemo(() => {
+    const list: {
+      fromToken: JSX.Element[];
+      toToken: JSX.Element[];
+      additional: JSX.Element[];
+    } = {
+      fromToken: [],
+      toToken: [],
+      additional: [],
+    };
+
+    if (fromTokenInfo) {
+      // is unknown
+      if (checkIsUnknownToken(fromTokenInfo)) {
+        list.fromToken.push(
+          <UnknownTokenSuggestion key={'unknown' + fromTokenInfo.address} tokenInfo={fromTokenInfo} />,
+        );
+      }
+
+      // lst token
+      if (lstApy && lstApy.apys[fromTokenInfo.symbol]) {
+        list.fromToken.push(
+          <LSTSuggestion key={'lst' + fromTokenInfo.symbol} apyInPercent={lstApy.apys[fromTokenInfo.symbol]} />,
+        );
+      }
+    }
+
+    if (toTokenInfo) {
+      // is unknown
+      if (checkIsUnknownToken(toTokenInfo)) {
+        list.toToken.push(<UnknownTokenSuggestion key={'unknown' + toTokenInfo.address} tokenInfo={toTokenInfo} />);
+      }
+
+      // lst token
+      if (lstApy && lstApy.apys[toTokenInfo.symbol]) {
+        list.fromToken.push(
+          <LSTSuggestion key={'lst' + toTokenInfo.symbol} apyInPercent={lstApy.apys[toTokenInfo.symbol]} />,
+        );
+      }
+    }
+
+    // Freeze authority, Permanent delegate, Transfer Tax
+    if (tokenMetadata && fromTokenInfo && toTokenInfo) {
+      const tokenExt1 = tokenMetadata[0] ? extractTokenExtensionsInfo(tokenMetadata[0]) : undefined;
+      const tokenExt2 = tokenMetadata[1] ? extractTokenExtensionsInfo(tokenMetadata[1]) : undefined;
+
+      // Freeze authority, Permanent delegate
+      const freeze: TokenInfo[] = [];
+      const permanent: TokenInfo[] = [];
+
+      (() => {
+        if (
+          tokenExt1?.freezeAuthority &&
+          FREEZE_AUTHORITY_IGNORE_LIST.includes(fromTokenInfo.address) === false && // Ignore bluechip like, USDC, USDT
+          (tokenExt1.freezeAuthority === SystemProgram.programId.toString()) === false // Ignore system program
+        ) {
+          freeze.push(fromTokenInfo); // Only mark non-strict token, so USDC, USDT, don't get marked
+        }
+
+        if (tokenExt1?.permanentDelegate) {
+          permanent.push(fromTokenInfo);
+        }
+      })();
```

Comment:
probably is my copy pasted code. do we need to use IIFE here?

Reply:
done, thanks